src-tauri/src/rpc.rs

```diff
@@ -131,12 +133,21 @@ pub async fn publish_ephemeral(
     topic: Topic,
     payload: serde_json::Value,
 ) -> Result<(), RpcError> {
-    debug!(
-        command.name = "publish_ephemeral",
-        "RPC request received"
-    );
-
+    debug!(command.name = "publish_ephemeral", "RPC request received");
     let payload = serde_json::to_vec(&payload)?;
     rpc.publish_ephemeral(&topic, &payload).await?;
     Ok(())
 }
+
+#[tauri::command]
+pub async fn upload_file(rpc: State<'_, Rpc>, app: AppHandle) -> Result<Option<Hash>, RpcError> {
+    debug!(command.name = "upload_file", "RPC request received");
+    match app.dialog().file().blocking_pick_file() {
+        Some(file_path) => {
+            let file_path = file_path.into_path().expect("parseable file path");
+            let file_hash = rpc.upload_file(file_path).await?;
+            Ok(Some(file_hash))
+        }
+        None => Ok(None),
+    }
+}
```

Comment:
I think we want to offer functionality to resize / compress images here, probably a ticket for the future?

Reply:
Additionally we probably want to filter the file selector to only show certain file types

Reply:
Yeh, these will be required features in the future, I'll open issues :+1: